syntax = "proto3";
package cosmos.staking.v1beta1;

// Src: https://github.com/cosmos/cosmos-sdk/blob/master/proto/cosmos/staking/v1beta1/tx.proto

import "coin.proto";

// MsgDelegate defines a SDK message for performing a delegation of coins
// from a delegator to a validator.
message MsgDelegate {
    string            delegator_address = 1;
    string            validator_address = 2;
    base.v1beta1.Coin amount            = 3;
}

// MsgUndelegate defines a SDK message for performing an undelegation from a
// delegate and a validator.
message MsgUndelegate {
    string            delegator_address = 1;
    string            validator_address = 2;
    base.v1beta1.Coin amount            = 3;
}

// MsgBeginRedelegate defines a SDK message for performing a redelegation
// of coins from a delegator and source validator to a destination validator.
message MsgBeginRedelegate {
    string            delegator_address     = 1;
    string            validator_src_address = 2;
    string            validator_dst_address = 3;
    base.v1beta1.Coin amount                = 4;
}
